### PR TITLE
snap-confine: is_running_on_classic_distribution() looks into os-release

### PR DESCRIPTION
**Pull request by mvo5**
The is_running_on_classic_distribution() is now looking at
/etc/os-release to figure out if it is running on classic or not.

### CODE REVIEW

File: cmd/libsnap-confine-private/classic-test.c [hunk 1]
```diff
@@ -20,4 +20,50 @@
...
+	g_assert_false(is_running_on_classic_distribution());
+	unlink("os-release.core");
+}
+
+const char *os_release_classic_with_long_line = ""
+    "NAME=\"Ubuntu\"\n"
+    "VERSION=\"17.04 (Zesty Zapus)\"\n"
+    "ID=ubuntu\n"
+    "ID_LIKE=debian\n"
+    "LONG=line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.line.";
```
Comment (zyga) [inline cmd/libsnap-confine-private/classic-test.c]: Thank you :)

File: cmd/libsnap-confine-private/cleanup-funcs.h [hunk 2]
```diff
@@ -27,6 +27,10 @@
 #include <sys/types.h>
 #include <dirent.h>
 
+// SC_CLEANUP will run the given cleanup function when the variable next
+// to it goes out of scope.
+#define SC_CLEANUP(n) __attribute__((cleanup(n)))
```
Comment (zyga) [inline cmd/libsnap-confine-private/cleanup-funcs.h]: Do I guess someone suggested this at the sprint? :-)

Comment (mvo5) [reply cmd/libsnap-confine-private/cleanup-funcs.h]: Not sure who you have in mind ;) But I added this, I find it (slightly) easier to read (but not much). I wish we could use __attribute__(cleanup(f)) to emulate a go defer. I think with some #define magic that is possible, however I am too lazy to look into it :)

Comment (zyga) [reply cmd/libsnap-confine-private/cleanup-funcs.h]: I'll have a look. I think Tyler asked me a while to do this but I never got to it. With this macro in place I'm more than willing to make the code uniformly using it.

File: cmd/libsnap-confine-private/classic.c [hunk 3]
```diff
@@ -1,16 +1,25 @@
...
 bool is_running_on_classic_distribution()
 {
-	// NOTE: keep this list sorted please
-	return false
-	    || access("/var/lib/dpkg/status", F_OK) == 0
-	    || access("/var/lib/pacman", F_OK) == 0
-	    || access("/var/lib/portage", F_OK) == 0
-	    || access("/var/lib/rpm", F_OK) == 0
-	    || access("/var/lib/eopkg", F_OK) == 0
-	    || access("/sbin/procd", F_OK) == 0;
```
Comment (stolowski) [inline cmd/libsnap-confine-private/classic.c]: :heart_eyes:

**Pull request merged**